### PR TITLE
Require admin privileges to view inspections

### PR DESCRIPTION
**Pull request by andchiind**

### CODE REVIEW

**Review by UsamaEquinorAFK — APPROVED** [review]
lgtm :)

**Comment by UsamaEquinorAFK** [comment]
Wait for Thomas before merging I would say, if he disagrees.

File: backend/api/Services/AccessRoleService.cs [hunk 1]
```diff
@@ -40,5 +40,18 @@ public async Task<List<string>> GetAllowedInstallationCodes(List<string> roles)
                     .Where((r) => roles.Contains(r.RoleName)).Select((r) => r.Installation != null ? r.Installation.InstallationCode.ToUpperInvariant() : "").ToListAsync();
             }
         }
+
+        public bool IsUserAdmin()
+        {
+            if (httpContextAccessor.HttpContext == null)
+                return false;
```
Comment (tsundvoll) [inline backend/api/Services/AccessRoleService.cs:47]:
```suggestion
            if not IsUserRequest()
                return false;
```

File: backend/api/Services/AccessRoleService.cs [hunk 2]
```diff
@@ -40,5 +40,18 @@ public async Task<List<string>> GetAllowedInstallationCodes(List<string> roles)
...
+            var roles = httpContextAccessor.HttpContext.GetRequestedRoleNames();
+            return roles.Contains(SUPER_ADMIN_ROLE_NAME);
+        }
+
+        public bool IsUserRequest()
```
Comment (tsundvoll) [inline backend/api/Services/AccessRoleService.cs:52]: Could this function name be more precise? What does it check?

File: backend/api/Services/InspectionFindingService.cs [hunk 3]
```diff
@@ -14,20 +14,24 @@ public async Task<List<InspectionFinding>> RetrieveInspectionFindings(DateTime l
             return inspectionFindings;
         }
 
-        public async Task<MissionRun?> GetMissionRunByIsarStepId(InspectionFinding inspectionFinding)
+        public async Task<MissionRun?> GetMissionRunByIsarStepId(string isarStepId)
```
Comment (tsundvoll) [inline backend/api/Services/InspectionFindingService.cs:17]: This is good; it takes in what the function "asks" for in its name

File: backend/api/Services/InspectionFindingService.cs [hunk 4]
```diff
@@ -14,20 +14,24 @@ public async Task<List<InspectionFinding>> RetrieveInspectionFindings(DateTime l
...
+#pragma warning restore CA1304
         }
 
-        public async Task<MissionTask?> GetMissionTaskByIsarStepId(InspectionFinding inspectionFinding)
+        public async Task<MissionTask?> GetMissionTaskByIsarStepId(string isarStepId)
         {
-            return await context.MissionTasks
-                .Where(missionTask => missionTask.Inspections.Any(inspection => inspection.IsarStepId == inspectionFinding.IsarStepId))
-                .FirstOrDefaultAsync();
+            var accessibleInstallationCodes = accessRoleService.GetAllowedInstallationCodes();
```
Comment (tsundvoll) [inline backend/api/Services/InspectionFindingService.cs:32]: is `accessibleInstallationCodes` variable used?

File: backend/api/Services/AccessRoleService.cs (hunk 2 above)
Comment (tsundvoll) [reply backend/api/Services/AccessRoleService.cs:52]:
```suggestion
        public bool IsAuthenticationAvailable()
```

**Pull request merged**